spyder_env_manager/spyder/widgets/packages_table.py

```diff
@@ -126,59 +112,110 @@ def row(self, row_num):
+        if packages:
+            if only_requested:
+                packages = list(filter(lambda package: package["requested"], packages))
+            for idx, package in enumerate(packages):
+                package["index"] = idx
+            packages_map = {package["name"]: package for package in packages}
+
+            self.source_model.packages = packages
+            self.source_model.packages_map = packages_map
+            self.source_model.reset()
+            self.resizeColumnToContents(NAME)
+
+    def next_row(self):
+        """Move to next row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.source_model.rowCount()
+        if row + 1 == rows:
+            row = -1
+        self.selectRow(row + 1)
+
+    def previous_row(self):
+        """Move to previous row from currently selected row."""
+        row = self.currentIndex().row()
+        rows = self.source_model.rowCount()
+        if row == 0:
+            row = rows
+        self.selectRow(row - 1)
+
     def contextMenuEvent(self, event):
         """Setup context menu"""
         row = self.rowAt(event.pos().y())
         packages = self.source_model.packages
-        if not packages[row]["dependence"]:
-            self.update_action = create_action(
-                self, _("Update package(s)"), triggered=self.selection
+        if packages and packages[row]["requested"]:
+            update_action = create_action(
+                self,
+                _("Update package"),
+                triggered=lambda: self.selection(
+                    EnvironmentPackagesActions.UpdatePackage, packages[row]
+                ),
             )
-            self.uninstall_action = create_action(
-                self, _("Uninstall package(s)"), triggered=self.selection
+            uninstall_action = create_action(
+                self,
+                _("Uninstall package"),
+                triggered=lambda: self.selection(
+                    EnvironmentPackagesActions.UninstallPackage, packages[row]
+                ),
             )
-            self.change_action = create_action(
+            change_action = create_action(
                 self,
-                _("Change package version with a version constraint"),
-                triggered=self.selection,
+                _("Change package version with a constraint"),
+                triggered=lambda: self.selection(
+                    EnvironmentPackagesActions.InstallPackageVersion, packages[row]
+                ),
             )
             menu = QMenu(self)
-            self.menu_actions = [
-                self.update_action,
-                self.uninstall_action,
-                self.change_action,
+            menu_actions = [
+                update_action,
+                uninstall_action,
+                change_action,
             ]
-            add_actions(menu, self.menu_actions)
+            add_actions(menu, menu_actions)
```

Comment:
`add_actions` is also low-level. Instead, you can use `self.create_menu` and then `self.add_item_to_menu`, both part of `SpyderWidgetMixin` too (see for instance how this is done in `spyder.widgets.browser.WebView`).